cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java

```diff
@@ -215,12 +215,13 @@ private PluginProperties substituteMacroFields(Plugin plugin, MacroEvaluator mac
     Map<String, String> properties = new HashMap<>();
     Map<String, PluginPropertyField> pluginPropertyFieldMap = plugin.getPluginClass().getProperties();
     MacroParser macroParser = new MacroParser(macroEvaluator);
-    for (Map.Entry<String, PluginPropertyField> pluginEntry : pluginPropertyFieldMap.entrySet()) {
-      if (pluginEntry.getValue().isMacroSupported()) {
-        String macroValue = plugin.getProperties().getProperties().get(pluginEntry.getKey());
-        properties.put(pluginEntry.getKey(), macroParser.parse(macroValue));
+
+    for (Map.Entry<String, String> property : plugin.getProperties().getProperties().entrySet()) {
+      PluginPropertyField field = pluginPropertyFieldMap.get(property.getKey());
+      if (field.isMacroSupported()) {
```

Comment:
its possible field is null if the user provided a property that is not in the PluginConfig. so this should check that field is not null and field is macro supported.